### PR TITLE
Enable e2e to run against either fake rp or real rp.

### PR DESCRIPTION
**Pull request by charlesakalugwu**
Also add support for the admin api

/cc @kargakis

### CODE REVIEW

File: pkg/fakerp/server.go [hunk 1]
```diff
@@ -19,11 +19,13 @@ import (
 	"github.com/ghodss/yaml"
 	"github.com/sirupsen/logrus"
 
-	"github.com/openshift/openshift-azure/pkg/api"
-	v20180930preview "github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	internalapi "github.com/openshift/openshift-azure/pkg/api"
+	externalapi "github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
```
Comment (kargakis) [inline pkg/fakerp/server.go]: Revert this change, I am adding support for the admin api which is also external.

**Comment by charlesakalugwu** [comment]
https://github.com/openshift/openshift-azure/issues/552

File: test/clients/azure/client.go [hunk 2]
```diff
@@ -19,12 +35,41 @@ func NewClientFromEnvironment() (*Client, error) {
...
+	focus := []byte(config.GinkgoConfig.FocusString)
+	switch {
+	case adminRpFocus.Match(focus), fakeRpFocus.Match(focus):
+		fmt.Println("configuring the fake resource provider")
+		rpURL = realfakerp.StartServer(log, conf, realfakerp.LocalHttpAddr)
+	case realRpFocus.Match(focus):
+		fmt.Println("configuring the real resource provider")
+		rpURL = externalapi.DefaultBaseURI
+	default:
+		panic(fmt.Sprintf("invalid focus %q - need to -ginkgo.focus=\\[Admin\\], -ginkgo.focus=\\[Fake\\] or -ginkgo.focus=\\[Real\\]", config.GinkgoConfig.FocusString))
```
Comment (jim-minter) [inline test/clients/azure/client.go]: doesn't this imply the Makefile needs to be updated?

File: test/clients/azure/client.go [hunk 3]
```diff
@@ -19,12 +35,41 @@ func NewClientFromEnvironment() (*Client, error) {
...
+
+	conf, err := realfakerp.NewConfig(log, true)
+	if err != nil {
+		return nil, err
+	}
+
+	var rpURL string
+	focus := []byte(config.GinkgoConfig.FocusString)
+	switch {
+	case adminRpFocus.Match(focus), fakeRpFocus.Match(focus):
```
Comment (jim-minter) [inline test/clients/azure/client.go]: the reason this is succeeding without the Makefile being updated is that this is matching the Admin in "[CustomerAdmin]" - I'd have expected you would be looking for "[someTag]" in square brackets and updating the Makefile?

**Comment by jim-minter** [comment]
this can go in now for expediency, but longer term I would prefer that the fake rp ran in a separate process which was set up via the test scripts, and the tests took in an argument which allowed overriding the RP endpoint to communicate with.

File: test/clients/azure/client.go (hunk 3 above)
Comment (kargakis) [reply test/clients/azure/client.go]: @charlesakalugwu please raise an issue about this

Comment (charlesakalugwu) [reply test/clients/azure/client.go]: #890
